api/watch/watch.go

```diff
@@ -42,6 +42,7 @@ type Plan struct {
 	client       *consulapi.Client
 	lastParamVal BlockingParamVal
 	lastResult   interface{}
+	QueryOptions consulapi.QueryOptions
```

Comment:
I feel including the entire `QueryOptions` is overblown. How about adding the following field to the `Plan struct`
`Namespace, Partition, Peer,` which should be sufficient?

Reply:
There are many parameters in the `QueryOptions`, and I was wondering if it would be inconvenient for users to add them one by one. If users need to set new configurations, they will need to continue adding specific configurations in the plan.